server/src/main/java/org/elasticsearch/bootstrap/FilePermissionUtils.java

```diff
@@ -63,25 +63,45 @@ public static void addSingleFilePath(Permissions policy, Path path, String permi
      */
     @SuppressForbidden(reason = "only place where creating Java-9 compatible FilePermission objects is possible")
     public static void addDirectoryPath(Permissions policy, String configurationName, Path path, String permissions) throws IOException {
+        addDirectoryPath(policy, configurationName, path, permissions, false);
+    }
+
+    /**
+     * Add access to path with direct and/or recursive access. This also creates the directory if it does not exist.
+     *
+     * @param policy            current policy to add permissions to
+     * @param configurationName the configuration name associated with the path (for error messages only)
+     * @param path              the path itself
+     * @param permissions       set of file permissions to grant to the path
+     * @param recursiveAccessOnly   indicates if the permission should provide recursive access to files underneath
+     */
+    @SuppressForbidden(reason = "only place where creating Java-9 compatible FilePermission objects is possible")
+    public static void addDirectoryPath(Permissions policy, String configurationName, Path path, String permissions,
```

Comment:
similar comment: can we please modify the few places calling the existing method instead of adding another variant?